vars/cloudFoundryDeploy.groovy

```diff
@@ -0,0 +1,161 @@
+import com.sap.piper.Utils
+import com.sap.piper.ConfigurationHelper
+
+import groovy.transform.Field
+
+@Field String STEP_NAME = 'cloudFoundryDeploy'
+@Field Set STEP_CONFIG_KEYS = [
+    'cloudFoundry',
```

Comment:
One idea of these lists was to get an good overview which parameters/config properties are available. 
The cloud foundry specific parameter keys are now hidden behind cloudFoundry and it is not visible which parameters are configurable here. (e.g. api, credentials, space, org, ...)
@marcusholl This might be a follow-up to our design decision to have these deep data structures. We should discuss, whether we also want to filter on this deep structures.

Reply:
> The cloud foundry specific parameter keys are now hidden

Yes, that's bad. One option is to support deep filtering, that means we have something like:

```
STEP_CONFIG_KEYS = [
   [aKey, [subLevelKey1,
           subLevelKey2],
    bKey, [] // no deep structure here
    ]
]
```

This is not very handy IMO.

Other approach is

```

ConfigurationHelper
           .loadStepDefaults(this)
           [...]
           .withMandatoryProperty('changeManagement/clientOpts')
           .withMandatoryProperty('changeManagement/credentialsId')
           .withMandatoryProperty('changeManagement/endpoint')
           .withMandatoryProperty('changeManagement/git/from')
           .withMandatoryProperty('changeManagement/git/to')
           .withMandatoryProperty('changeManagement/git/format')
           .withMandatoryProperty('developmentSystemId')

```

I use this approach [here](https://github.com/SAP/jenkins-library/blob/5b8dd91510d1e98114b6348eeff77bc19af87060/vars/transportRequestCreate.groovy#L31). I this this is an alternate suitable approach for checking paramters.

I understand the difference between the approaches: with the old approach we filter parameters so that they are not availble to the step. Hence they cannot used by accident. This is a high value. With the appraoch outlined above we do not filter, we only ensure that the mentioned parameters are present.